core/src/main/java/org/apache/iceberg/SystemConfigs.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg;
+
+import java.util.function.Function;
+
+/**
+ * Configuration properties that are controlled by Java system properties or environmental variable.
+ */
+public class SystemConfigs {
+
+  private SystemConfigs() {}
+
+  /**
+   * Sets the size of the worker pool. The worker pool limits the number of tasks concurrently
+   * processing manifests in the base table implementation across all concurrent planning or commit
+   * operations.
+   */
+  public static final ConfigEntry<Integer> WORKER_THREAD_POOL_SIZE =
+      new ConfigEntry<>(
+          "iceberg.worker.num-threads",
+          "ICEBERG_WORKER_NUM_THREADS",
+          Math.max(2, Runtime.getRuntime().availableProcessors()),
+          Integer::parseUnsignedInt);
```

Comment:
I think previously it would throw an exception if invalid, and return default.  But now it may not do so?

Reply:
Good catch, updated.